app/models/course.rb

```diff
@@ -3,5 +3,20 @@ class Course < ApplicationRecord
   belongs_to :trimester
   has_many :enrollments
 
-  delegate :title, to: :coding_class
+   def student_name_list
+    names_list = []
+    self.enrollments.each do |enrollment|
```

Comment:
```suggestion
    enrollments.each do |enrollment|
```
nit: you can remove self